collie/model/base/base_pipeline.py

```diff
@@ -179,14 +179,27 @@ def __init__(self,
                     f'{self.train_loader.num_items} != {self.val_loader.num_items}.'
                 )
 
+                if (
+                    hasattr(self.train_loader, 'negative_sample_type')
+                    or hasattr(self.val_loader, 'negative_sample_type')
+                ):
+                    assert self.train_loader.negative_sample_type == \
+                        self.val_loader.negative_sample_type, (
+                            'Training and val ``negative_sample_type`` property must both equal '
+                            '``item`` or both equal ``user``, not: '
+                            f'{self.train_loader.negative_sample_type}'
+                            f' and {self.val_loader.negative_sample_type}, respectively.'
```

Comment:
I think it's okay to just say something along the lines of: 

```suggestion
                            'Training and val ``negative_sample_type`` property must both equal: '
                            f'{self.train_loader.negative_sample_type} != '
                            f'{self.val_loader.negative_sample_type}.'
```